jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.

```

Comment:
Please follow numpydoc convention: https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

@kecnry , does this class need to be exposed in API doc on ReadTheDocs?

Reply:
I don't think at this point... when we get to the editable spline effort we'll have to see where we want that API access to be exposed.